src/AzureAppConfigurationImpl.ts

```diff
@@ -196,17 +240,65 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
             keyValues.push([key, value]);
         }
 
-        this.#configMap.clear(); // clear existing key-values in case of configuration setting deletion
+        this.#clearLoadedKeyValues(); // clear existing key-values in case of configuration setting deletion
         for (const [k, v] of keyValues) {
             this.#configMap.set(k, v);
         }
     }
 
+    async #clearLoadedKeyValues() {
+        for (const key of this.#configMap.keys()) {
+            if (key !== FEATURE_MANAGEMENT_KEY_NAME) {
+                this.#configMap.delete(key);
+            }
+        }
+    }
+
+    async #loadFeatureFlags() {
+        // Temporary map to store feature flags, key is the key of the setting, value is the raw value of the setting
+        const featureFlagsMap = new Map<string, any>();
+        for (const selector of this.#featureFlagSelectors) {
+            const listOptions: ListConfigurationSettingsOptions = {
+                keyFilter: `${featureFlagPrefix}${selector.keyFilter}`,
+                labelFilter: selector.labelFilter
+            };
+
+            const pageEtags: string[] = [];
+            const pageIterator = listConfigurationSettingsWithTrace(
+                this.#requestTraceOptions,
+                this.#client,
+                listOptions
+            ).byPage();
+            for await (const page of pageIterator) {
+                if (page._response.status === 200) {
+                    if (page.etag) {

```

Comment:
This check covers null or empty both? If the etag is null or empty, I think we should still push it into the list of page etags. This should never happen, but if it does happen due to some server bug, we need to preserve the order of server returned page etags. Otherwise, the next request to load FF using this list of page etags would be indeterministic.

Reply:
It makes sense to keep the order, I'll remove the check here, and the 200 status code check above.

More info:
In SDK, page.etag defintion as below, can be either `string` or `undefined`. 
It cannot be `null`, otherwise it would be a bug of SDK.

```ts
/**
 * Entity with etag.
 */
export declare interface EtagEntity {
    /**
     * The etag for this entity
     */
    etag?: string;
}
```

the `if(page.etag)` check is false if `page.etag` is undefined or null or empty string or even a number `0`. But given the type definition, the possible/legal values are string or undefined. So the check is identical to "isUndefinedOrEmptyString".

Reply:
BTW, as SDK defines `pageEtags` as `string[]`, meaning each element of the pageEtags list can only be of `string` type. Thus we cannot put `undefined` as a placeholder. I would use an empty string for the case when pageEtag of the response is null or empty.

Reply:
3c12654